### ISSUE TITLE
[Bug] adblocker doesn't check if request is already intercepted

### ISSUE THREAD

**Issue by GlenDC** [berstend/puppeteer-extra#600]
First create a custom plugin which intercepts certain domains or resources, e.g.:

```javascript
'use strict'

const {
    PuppeteerExtraPlugin
} = require('puppeteer-extra-plugin')

const domains = [
    "google-analytics.com",
   // ...
];
const blockedRegExp = new RegExp('(' + domains.join('|') + ')', 'i');

class Plugin extends PuppeteerExtraPlugin {
    constructor(opts = {}) {
        super(opts)
    }

    get name() {
        return 'firewall-domains'
    }

    get defaults() {
        return {}
    }

    async onPageCreated(page) {
        page.on('request', async (request) => {
            if (request.isInterceptResolutionHandled()) {
                return;
            }

            const url = request.url();
            if (blockedRegExp.test(url)) {
                return await request.abort();
            }
        });
    }
}

module.exports = function (pluginConfig) {
    return new Plugin(pluginConfig)
}
```

Use that plugin as well as the puppeteer-extra-plugin-adblocker (`blockTrackers: true`),
and visit a page such as https://www.whatsmyua.info.

You'll get a "Request is already handled" exception.

```javascript
const puppeteer = require('puppeteer-extra')

const AdblockerPlugin = require('puppeteer-extra-plugin-adblocker');
puppeteer.use(AdblockerPlugin({
    blockTrackers: true
}));

puppeteer.use(require('./plugins/mine')());

(async () => {
    const browser = await puppeteer.launch()
    const page = await browser.newPage();

    await page.goto(args.url, {
        waitUntil: 'domcontentloaded'
    });
    await page.screenshot({
        path: './my-screenshot.png',
        fullPage: true
    });
})()
```

Adding the following at the start of `this.onRequest` seems to remove the error:

```
if (details.isInterceptResolutionHandled()) {
     return;
}
```

Is this normal? Desired? Seems like it must be a common thing with this kind of plugin ecosystem that multiple request handles handle the same request for different request. If so, what am I missing? Or is this really just a bug? Would surprise me though.

**Versions**

```
  System:
    OS: macOS 11.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 54.82 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.18.3 - ~/.nvm/versions/node/v12.18.3/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v12.18.3/bin/yarn
    npm: 7.23.0 - ~/.nvm/versions/node/v12.18.3/bin/npm
  npmPackages:
    puppeteer: ^13.0.1 => 13.0.1 
    puppeteer-extra: ^3.2.3 => 3.2.3 
    puppeteer-extra-plugin-adblocker: ^2.12.0 => 2.12.0 
    puppeteer-extra-plugin-anonymize-ua: ^2.3.3 => 2.3.3 
    puppeteer-extra-plugin-stealth: ^2.9.0 => 2.9.0 
```

**Comment by josep11** [comment]
I just found the same issue on intercepting images.

**Comment by b5414** [comment]
same

**Comment by Hudsonzp** [comment]
Yup, happens to me as well.

**Comment by berstend** [comment]
This has been fixed through #592 :)
